src/state/slices/foxEthSlice/foxEthSlice.ts

```diff
@@ -0,0 +1,353 @@
+import { Contract } from '@ethersproject/contracts'
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/dist/query/react'
+import { ethAssetId } from '@shapeshiftoss/caip'
+import { Fetcher, Token } from '@uniswap/sdk'
+import IUniswapV2Pair from '@uniswap/v2-core/build/IUniswapV2Pair.json'
+import dayjs from 'dayjs'
+import {
+  foxAssetId,
+  foxEthLpAssetId,
+  UNISWAP_V2_WETH_FOX_POOL_ADDRESS,
+} from 'features/defi/providers/fox-eth-lp/constants'
+import farmAbi from 'features/defi/providers/fox-farming/abis/farmingAbi.json'
+import {
+  calculateAPRFromToken0,
+  getEthersProvider,
+  makeTotalLpApr,
+  rewardRatePerToken,
+} from 'plugins/foxPage/utils'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { logger } from 'lib/logger'
+import { marketData } from 'state/slices/marketDataSlice/marketDataSlice'
+
+import { FOX_TOKEN_CONTRACT_ADDRESS, WETH_TOKEN_CONTRACT_ADDRESS } from './constants'
+import {
+  farmOpportunities,
+  FoxEthLpEarnOpportunityType,
+  FoxFarmingEarnOpportunityType,
+  lpOpportunity,
+} from './foxEthCommon'
+import { fetchPairData } from './utils'
+
+type FoxEthState = {
+  farmOpportunities: FoxFarmingEarnOpportunityType[]
+  lpOpportunity: FoxEthLpEarnOpportunityType
+}
+
+const initialState: FoxEthState = {
+  farmOpportunities,
+  lpOpportunity,
+}
+
+const moduleLogger = logger.child({ namespace: ['foxEthSlice'] })
+
+export const foxEth = createSlice({
+  name: 'foxEth',
+  initialState,
+  reducers: {
+    clear: () => initialState,
+    upsertLpOpportunity: (state, action: PayloadAction<Partial<FoxEthLpEarnOpportunityType>>) => {
+      state.lpOpportunity = {
+        ...state.lpOpportunity,
+        ...action.payload,
+      }
+    },
+    upsertFarmingOpportunity: (
+      state,
+      action: PayloadAction<Partial<FoxFarmingEarnOpportunityType>>,
+    ) => {
+      const stateOpportunityIndex = state.farmOpportunities.findIndex(
+        opportunity => opportunity.contractAddress === action.payload.contractAddress,
+      )
+      state.farmOpportunities[stateOpportunityIndex] = {
+        ...state.farmOpportunities[stateOpportunityIndex],
+        ...action.payload,
+      }
+    },
+  },
+})
+
+type getFoxEthLpGeneralDataReturn = {
+  tvl: string
+  apy: string
+  isLoaded: boolean
```

Comment:
We won't need this anymore, we can use RTK's `isLoading` and `data` fields to check for data loading/presence